tests/unit/game/effects/weakness.test.ts

```diff
@@ -0,0 +1,320 @@
+					).toBe(VintageBeefCommon.health - ImpulseSVCommon.primary.damage)
+				},
+			},
+			{startWithAllCards: true, noItemRequirements: true},
+		)
+	})
+
+	test('Weakness Self Redundancy', () => {
+		testGame(
+			{
+				playerOneDeck: [VintageBeefCommon],
+				playerTwoDeck: [VintageBeefCommon, PotionOfWeakness, PotionOfWeakness],
+				saga: function* (game) {
+					yield* playCardFromHand(game, VintageBeefCommon, 'hermit', 0)
+					yield* endTurn(game)
+
+					yield* playCardFromHand(game, VintageBeefCommon, 'hermit', 0)
+					yield* playCardFromHand(game, PotionOfWeakness, 'single_use')
+					yield* endTurn(game)
+
+					yield* endTurn(game)
+
+					yield* playCardFromHand(game, PotionOfWeakness, 'single_use')
+					yield* attack(game, 'primary')
+					yield* endTurn(game)
+
+					expect(
+						game.components.find(
+							RowComponent,
+							query.row.currentPlayer,
+							query.row.index(0),
+						)?.health,
+					).toBe(
+						VintageBeefCommon.health - VintageBeefCommon.primary.damage - 20,
+					)
+				},
+			},
+			{startWithAllCards: true, noItemRequirements: true},
+		)
+	})
+
+	test('Weakness Heteroredundancy', () => {
+		testGame(
+			{
+				playerOneDeck: [SmallishbeansRare],
+				playerTwoDeck: [ImpulseSVCommon, PotionOfWeakness],
+				saga: function* (game) {
+					yield* playCardFromHand(game, SmallishbeansRare, 'hermit', 0)
+					yield* endTurn(game)
+
+					yield* playCardFromHand(game, ImpulseSVCommon, 'hermit', 0)
+					yield* playCardFromHand(game, PotionOfWeakness, 'single_use')
+					yield* attack(game, 'primary')
+					yield* endTurn(game)
+
+					expect(
+						game.components.find(
+							RowComponent,
+							query.row.currentPlayer,
+							query.row.index(0),
+						)?.health,
+					).toBe(SmallishbeansRare.health - ImpulseSVCommon.primary.damage - 20)
+				},
+			},
+			{startWithAllCards: true, noItemRequirements: true},
+		)
+	})
+
+	test('Weakness Lethality Check', () => {
+		testGame(
+			{
+				playerOneDeck: [ImpulseSVCommon, SmallishbeansRare, ImpulseSVCommon],
+				playerTwoDeck: [
+					SmallishbeansRare,
+					SmallishbeansRare,
+					SmallishbeansRare,
+					PotionOfWeakness
+				],
+				saga: function* (game) {
+					yield* playCardFromHand(game, ImpulseSVCommon, 'hermit', 0)
+					yield* playCardFromHand(game, SmallishbeansRare, 'hermit', 1)
+					yield* playCardFromHand(game, ImpulseSVCommon, 'hermit', 2)
+					yield* endTurn(game)
+
+					yield* playCardFromHand(game, SmallishbeansRare, 'hermit', 0)
+					yield* playCardFromHand(game, SmallishbeansRare, 'hermit', 1)
+					yield* playCardFromHand(game, SmallishbeansRare, 'hermit', 2)
+					yield* attack(game, 'secondary')
+					yield* endTurn(game)
+
+					yield* endTurn(game)
+
+					yield* playCardFromHand(game, PotionOfWeakness, 'single_use')
+					yield* attack(game, 'secondary')
+					yield* endTurn(game)
+
+					yield* pick(
+						game,
+						query.slot.currentPlayer,
+						query.slot.hermit,
+						query.slot.rowIndex(1),
+					)
```

Comment:
This is a regular change active row action, not from using chorus fruit
```suggestion
					yield* changeActiveHermit(game, 1)
```